### PR TITLE
Dave myriad improvements

### PR DESCRIPTION
**Pull request by daveauerbach**
Add statsd
Allows for JSON headers
Uniform serialization for stream data
Better serialization for stream data

### CODE REVIEW

File: cfg/requirements.txt [hunk 1]
```diff
@@ -8,3 +8,4 @@ yapf
 msgpack-python
 python-snappy
 pyzmq
+pystatsd
```
Comment (pulltab) [inline cfg/requirements.txt]: Try this: 

```-e git://git@github.com/postmates/pystatsd.git@2.0.0#egg=pystatsd``

File: bin/tritond [hunk 2]
```diff
@@ -211,11 +222,12 @@ def main():
                 'PartitionKey': partition_key
             })
 
+        time_delta_ms = (time.time() - last_write) * 1000
```
Comment (pulltab) [inline bin/tritond]: You can use the new timing decorators here.

File: triton/stream.py [hunk 3]
```diff
@@ -234,7 +236,12 @@ def _select_shard_ids(self, shard_nums):
         return shard_ids
 
     def put(self, **kwargs):
-        data = msgpack.packb(kwargs)
+        try:
+            data = msgpack.packb(kwargs)
+        except TypeError:
+            # If we fail to serialize our context, we can try again with an
+            # enhanced packer (it's slower though)
```
Comment (MaCaki) [inline triton/stream.py]: a lot slower?   Would it be preferable to have the user cast their data into floats, or avoid using decimals all together if they expect or need high velocity writes?

**Comment by MaCaki** [comment]
I ran this branch in dev with pegasus and it worked. However I couldn't get it to install on my mac for local-local testing.

File: triton/stream.py (hunk 3 above)
Comment (daveauerbach) [inline triton/stream.py]: I couldn't imagine more than a 10% penalty; haven't benchmarked (this comment is from original author). Forcing serialization to users has lead to much confusion and some lost data.

**Comment by daveauerbach** [comment]
Best way to test this is to run the tests in py-base. I'll update test documentation.

**Comment by MaCaki** [comment]
passed 
r+

**Pull request merged**